pkg/operator/ceph/cluster/osd/envs.go

```diff
@@ -213,6 +213,19 @@ func osdActivateEnvVar() []v1.EnvVar {
 	return append(cephVolumeEnvVar(), monEnvVars...)
 }
 
+func getEnvFromSources() []v1.EnvFromSource {
+	optionalConfigMapRef := true
+
+	return []v1.EnvFromSource{
+		{
+			ConfigMapRef: &v1.ConfigMapEnvSource{
```

Comment:
What if the configmap does not exist? Will this fail or will it be ignored?

Reply:
The intended behavior is to ignore. [This is what the documentation of `ConfigMapEnvSource` says about the `Optional`](https://pkg.go.dev/k8s.io/api/core/v1#ConfigMapEnvSource):

```golang
type ConfigMapEnvSource struct {
	// ...
	// Specify whether the ConfigMap must be defined
	// +optional
	Optional *bool `json:"optional,omitempty" protobuf:"varint,2,opt,name=optional"`
}
```